controllers/healthcheck_controller.go

```diff
@@ -107,6 +109,31 @@ func (r *HealthCheckReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		}
 		return ctrl.Result{}, ignoreNotFound(err)
 	}
+
+	return r.execHealthCheck(ctx, req, log, healthCheck)
+}
+
+func (r *HealthCheckReconciler) execHealthCheck(ctx context.Context, req ctrl.Request, log logr.Logger, healthCheck *activemonitorv1alpha1.HealthCheck) (ctrl.Result, error) {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Info("Error: Panic occurred during execAdd %s/%s due to %s", healthCheck.Name, err)
+		}
+	}()
+	// Process HealthCheck
+	ret, procErr := r.processHealthCheck(ctx, req, log, healthCheck)
+
+	err := r.Update(ctx, healthCheck)
+	if err != nil {
+		log.Error(err, "Error updating healthcheck resource")
+		// Force retry when status fails to update
+		return ctrl.Result{}, err
+	}
+
+	return ret, procErr
+}
+
+func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl.Request, log logr.Logger, healthCheck *activemonitorv1alpha1.HealthCheck) (ctrl.Result, error) {
```

Comment:
I like the addition of recovering in a defer block with the addition of function wrapping; that looks great.

However, I think the names of the two newly added functions could be honed a bit. I think it's not entirely clear based on the names of `execHealthCheck` and `processHealthCheck` which one should call the other or how they are different.

But something like `processOrRecoverFromHealthCheck` calling `processHealthCheck` would somewhat make clear which is a superset of the other (and therefore the "outer" fn). I don't necessarily think the names I just came up with are perfect, so def still feel free to consider if you have some better ideas there. But the basic idea of the two names being slightly more clear WRT which is which, does seem like a good modification to make.

Reply:
sure makes sense I will make this change accordingly